### ISSUE TITLE
[BUG] Cannot match on `METHOD_CALL`s when the method is inherited

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#1044]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

latest

### Priority

Critical

### Current Behavior

When trying to match a `METHOD_CALL` for a method that is inherited, for instance:
```
public class MyMap extends HashMap<String, String> { ... }
```
and
```
location: METHOD_CALL
pattern: MyMap.put
```
the rule won't match.

### Expected Behavior

We should be able to match on inherited methods, which are quite common in Java frameworks.

### How Reproducible

Always (Default)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by shawn-hurley** [comment]
@jmle, I am curious if this is still an issue? 

I am going to unmark stale because I don't know, but wondering if you know off the top of your head.

**Comment by jmle** [comment]
@shawn-hurley yes, this is still happening, but I haven't taken a look at it yet
